### DISCUSSION TITLE
Quick Support and configuration

### DISCUSSION THREAD

**Discussion by pdifeo** [rustdesk/rustdesk#5618] [Q&A] [answered]
Hi,

can someone help me how to generate a quicksupport executable with embedded configuration to my own rustdesk server pro ?

Thnaks
Pasquale

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/4903

**Reply by rustdesk** [reply]
https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/console/#windows-exe-is-where-the-configs-are-shown

**Comment by pdifeo** [comment]
@rustdesk I know . But 

- If I rename original executable with qs at the end, rustdesk client ask me for user elevation
- If I rename original executable with the configuration in the name, rustdesk don't permit me to do user elevation
- If I rename original executable with the configuration in the name and qs at the end ... client ask me for user elevation BUT don't take the configuration

The test was executed on workstation where never rustdesk was executed in the past. Otherwise, it remember previous configurations

**Reply by rustdesk** [reply]
Need some code change (will do soon), you are the first one for this usage.

**Reply by rustdesk** [reply]
@21pages

**Reply by pdifeo** [reply]
@rustdesk  Strange that I am the first to ask this. If I bought the licence to have a faster server, it is useful for me to also have quicksupport on my own server and not on the public ones.
Thank you in advance for your interest.
Pasquale

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/commit/dc29b4afa16088d4c3c4474813ebcfda6419c828, will be ready soon (the nightly build)

**Reply by rustdesk** [reply]
Name it like this `rustdesk-qs-licensed-fsfsfsdfesfes.exe`, the file name containing `-qs-`.

**Reply by pdifeo** [reply]
OK, I will test as soon as possible

**Reply by rustdesk** [reply]
New nightly build is ready now.

**Reply by pdifeo** [reply]
It seems to be working. But I want to do some 'clean' tests. I will do them in the evening Italian time

**Reply by dinger1986** [reply]
That's brilliant thanks @rustdesk

**Reply by bilogic** [reply]
does `-qs-` work with macos DMG?

**Comment by dinger1986** [comment]
hows this expected to work?

I downloaded the latest nightly build 2 hours ago it was built, renamed to `rustdesk-qs-licensed-myrandomstring.exe

Ran it and go this:

![image](https://github.com/rustdesk/rustdesk/assets/7244447/ee3d559c-9e42-4e7d-b859-d024e00ba4af)

It seems to be the same as if it doesnt have -qs- in it

**Comment by pdifeo** [comment]
On your client can ask to elevate the privileges on action menu. Otherwise
you cannot do

**Reply by dinger1986** [reply]
Yeah that's fine, at a minimum it should hide the install option. Would save a lot of confusion

**Reply by pdifeo** [reply]
No must continue to be there. This whole flow allows you to remotely install the client on the computer. Otherwise it is a mess.

**Reply by dinger1986** [reply]
But if you install does it copy the settings across correctly or just use the public servers after?

**Reply by pdifeo** [reply]
Copy the settings correctly

**Comment by rustdesk** [comment]
No need to discuss about this, we know what to do with `quick support` version， have done a lot of research. The only issue is we need `time` for dev. Right now, this is not the highest prioirty, behind shared adressbook, hardware codecs, preventing scammer, mulitple monitor etc.

**Reply by dinger1986** [reply]
I'm only asking what's been changed cause it looks to me like the qs flag isn't doing anything now

**Comment by rustdesk** [comment]
@21pages let's target to release a real `-qs-` in 1.2.4 or 1.2.5, hide the install option / display tab under settings. How to present the UI after removing below parts? If you use small width, how to deal with settings page?

and configurable like this, https://github.com/rustdesk/rustdesk/discussions/6668#discussioncomment-8407535

![image](https://github.com/rustdesk/rustdesk/assets/71636191/6fd1e604-b406-4a67-ae9f-27ece8386b88)

**Reply by 21pages** [reply]
I think we can follow teamviewer and make the settings as simple as possible.
![1697362275531](https://github.com/rustdesk/rustdesk/assets/14891774/25284b1b-ad6b-44fb-8b5a-d7342644598f)

**Reply by rustdesk** [reply]
Let's do it

**Reply by rustdesk** [reply]
Priority lower than shared addiress book and hwcodecs.

**Reply by Sebshino** [reply]
hello, The quicksupport version should not populate the address book because quicksupport is often used occasionally.  it would also be necessary to be able to integrate server information without the client having access to it in order to be able to offer it on our websites.  Teamviewer offers a quicksupport client generator to integrate a default password, integrate a personalized logo, etc.

**Reply by NB516** [reply]
Looking forward to testing a real -qs- client.

**Reply by err0r2k** [reply]
Me too. Any news about this?

**Reply by strongwazz** [reply]
Would be very interested in this :)

**Reply by HollaTheWoodFairy** [reply]
I'm also awaiting this feature :-) Thanks for your hard work @rustdesk

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/discussions/8981#discussioncomment-10263847
